src/commands/periscope/periscope.ts

```diff
@@ -67,47 +70,57 @@ async function runAKSPeriscope(
+    }
 
-    if (!aksDeploymentFile) return undefined;
+    const kustomizeConfig = getKustomizeConfig();
+    if (failed(kustomizeConfig)) {
+        vscode.window.showErrorMessage(kustomizeConfig.error);
+        return undefined;
+    }
 
-    const runCommandResult = await longRunning(`Running AKS Periscope on ${clusterName}.`,
-        () => runAssociatedAKSPeriscopeCommand(aksDeploymentFile, clusterKubeConfig)
+    const aksDeploymentFile = await longRunning(`Creating AKS Periscope resource specification for ${clusterName}.`,
+        () => prepareAKSPeriscopeKustomizeOverlay(clusterStorageInfo.result, kustomizeConfig.result)
     );
 
-    await createPeriscopeWebView(clusterName, runCommandResult, clusterStorageInfo);
-
-}
+    if (failed(aksDeploymentFile)) {
+        vscode.window.showErrorMessage(aksDeploymentFile.error);
+        return undefined;
+    }
 
-async function runAssociatedAKSPeriscopeCommand(
-    aksPeriscopeFile: string,
-    clusterKubeConfig: string | undefined
-): Promise<k8s.KubectlV1.ShellResult | undefined> {
-    const kubectl = await k8s.extension.kubectl.v1;
+    const runCommandResult = await longRunning(`Deploying AKS Periscope to ${clusterName}.`,
+        () => deployKustomizeOverlay(kubectl, aksDeploymentFile.result, clusterKubeConfig)
+    );
 
-    if (!kubectl.available) {
-        vscode.window.showWarningMessage(`Kubectl is unavailable.`);
+    if (failed(runCommandResult)) {
+        vscode.window.showErrorMessage(runCommandResult.error);
         return undefined;
     }
 
-    // Clean up running instance.
-    await tmpfile.withOptionalTempFile<k8s.KubectlV1.ShellResult | undefined>(
-        clusterKubeConfig, "YAML",
-        (f) => kubectl.api.invokeCommand(`delete ns aks-periscope --kubeconfig="${f}"`));
-
-    // Deploy the aks-periscope.
-    const runCommandResult = await tmpfile.withOptionalTempFile<k8s.KubectlV1.ShellResult | undefined>(
-        clusterKubeConfig, "YAML",
-        (f) => kubectl.api.invokeCommand(`apply -f ${aksPeriscopeFile} --kubeconfig="${f}" && kubectl cluster-info --kubeconfig="${f}"`));
+    await createPeriscopeWebView(clusterName, runCommandResult.result, clusterStorageInfo.result);
+}
 
-    return runCommandResult;
+async function deployKustomizeOverlay(
+    kubectl: k8s.APIAvailable<k8s.KubectlV1>,
+    overlayDir: string,
+    clusterKubeConfig: string
+): Promise<Errorable<k8s.KubectlV1.ShellResult>> {
+    return await tmpfile.withOptionalTempFile<Errorable<k8s.KubectlV1.ShellResult>>(clusterKubeConfig, "YAML", async kubeConfigFile => {
+        // Clean up running instance (without an error if it doesn't yet exist).
+        const deleteResult = await invokeKubectlCommand(kubectl, kubeConfigFile, 'delete ns aks-periscope --ignore-not-found=true');
+        if (failed(deleteResult)) return deleteResult;
+
+        // Deploy aks-periscope.
+        const applyResult = await invokeKubectlCommand(kubectl, kubeConfigFile, `apply -k ${overlayDir}`);
```

Comment:
💡- Keen to mention that we should make sure `-k` option don't have dependency on `kustomise` https://kustomize.io i..e currently we have min version of `kubectl` from parent extension and if for -k flag we need `customise` pre-baked from that route for host will not have this.

I thought I will share. Either way if that is the case we have other options handy for this to deliver but worth checking. Thanks heaps.

Reply:
Great point - I've noted the minimum `kubectl` version in the main PR description.